component/componentattribute/logger.go

```diff
@@ -0,0 +1,52 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package componentattribute // import "go.opentelemetry.io/collector/component/componentattribute"
+
+import (
+	"go.opentelemetry.io/otel/attribute"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var _ zapcore.Core = (*Core)(nil)
+
+type Core struct {
+	zapcore.Core
+	from  *zap.Logger
+	attrs attribute.Set
+}
```

Comment:
Looking at usages (outside the service) the only think needed is `Without(keys ...string) *zap.Logger`

Can we do some golang magic, and actually public expose only an interface (rest of the implementation is in otelcol/service)?

Reply:
Here is where I got https://github.com/open-telemetry/opentelemetry-collector/pull/12289

I don't have to expose public any interface. We will still need to expose the fields keys not sure where though.

Reply:
> I don't have to expose public any interface

I'm trying to incorporate your solution to simplify mine, but one consideration overlooked is that in practice we pass different kinds of loggers into tests, so we must make a type assertion to check for the `Without` method. If we're doing this, I think we need to at least give a name the interface so that components do not need to assert against an anonymous interface. The alternative would be to enforce somehow that tests use the same logger, but this is impractical IMO.

Reply:
Similarly, for tests that do want to validate the functionality of the shared logger, they need to be able to instantiate a logger, so at least then you need to expose a testing logger which is very similar to the actual logger.

Reply:
I was able to reduce the surface area of `componentattribute` but still think we need a constructor for the logger